notifications-worker.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/dxe/alc-mobile-api/expo"
+
+	"github.com/dxe/alc-mobile-api/model"
+
+	"github.com/jmoiron/sqlx"
+)
+
+func NotificationsWorker(db *sqlx.DB) {
+	log.Println("Notifications Worker started.")
+
+	tx, err := db.Beginx()
+	if err != nil {
+		log.Println("Failed to begin database transaction.")
+		return
+	}
+
+	notifications, err := model.SelectNotificationsToSend(tx)
+	if err != nil {
+		log.Printf(err.Error())
+		tx.Rollback()
+		return
+	}
+
+	if len(notifications) == 0 {
+		log.Println("Found no queued notifications to send.")
+		tx.Rollback()
```

Comment:
Is this an error state? Would it be clearer to commit nothing?

Reply:
It is not an error state. It just means that there are notifications that need sending. I'm not sure what you mean by "commit nothing." Do you mean just skip the rollback b/c nothing would have been selected?

Reply:
I think Alexis means that since it's not a failure, you could just tx.Commit() here instead. Since no side-effects have happened, I don't think there should be any actual difference though.

Reply:
Ah, that makes sense.